### ISSUE TITLE
Add compute.backendServices.list to the list of permissions required for workload identity sa

### ISSUE THREAD

**Issue by arya-harness** [GoogleCloudPlatform/gke-autoneg-controller#138]
Hi Team

I am getting a
```
ERROR	Reconciler error	{"controller": "service", "controllerGroup": "", "controllerKind": "Service", "Service": {"name":"******","namespace":"*******"}, "namespace": "******", "name": "********r", "reconcileID": "cf4d3868-3fea-4e22-9c63-939feac01985", "error": "googleapi: Error 403: Required 'compute.backendServices.list' permission for 'projects/*********', forbidden"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.1/pkg/internal/controller/controller.go:324
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.1/pkg/internal/controller/controller.go:265
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.1/pkg/internal/controller/controller.go:226

```
Seems to be benign here mostly but unsure if it is actually needed.
Version: `v1.0.0`

A suggestion from my end would be to update the script [here](https://github.com/GoogleCloudPlatform/gke-autoneg-controller/blob/master/deploy/workload_identity.sh) if its needed.

Thanks

**Comment by rosmo** [comment]
You're right, that script is severely outdated - I need to update it, but I would strongly recommend using another installation method than manual (even a simple Terraform set up should be quicker than manual installation).

**Comment by arya-harness** [comment]
Could be, havent found the relevant change in NEG side yet.
On a seperate note or slightly related, if we change the annotation of the service to a different neg name, AutoNEG does create the new NEG, but it does not delete the older one.
It only does deletion, when the service is deleted(probably due to the finalizer trigger). 
I wonder if it is the intended behaviour or there is a lacking permission on AutoNEG side to delete it on the fly?

**Comment by rosmo** [comment]
Autoneg does not create or change NEGs, that's handled through the `cloud.google.com/neg` annotation and another controller. Autoneg just adds the created NEGs to backend services.

**Comment by arya-harness** [comment]
My apologies, I meant that flow itself. AutoNEG is not deattaching the NEG from the GCP backend when the name of NEG is changed.

**Comment by rosmo** [comment]
Maybe you could outline the steps that have to happen for you to hit this bug?

**Comment by arya-harness** [comment]
Following up on this thread a bit late, I found this issue while I was changing the NEG name. 
Steps to reproduce:
- Create Backend Services(Global/Regional),
- Create a Kubernetes service annotate it with autoNeg annotations
- Edit the name of NEG and observe.

---
Examples:
- Created Regional backend
- Added annotations
```
cloud.google.com/neg: '{"exposed_ports": {"80":{"name": "gateway-neg"}}}'
controller.autoneg.dev/neg: {"backend_services":{"80":[{"name":"internal-backend","region":"us-west1","max_rate_per_endpoint":200}]}}
 ```
- Let it sync in and create NEGs. This creates 3 Network Endpoints for all 3 regions named as gateway-neg.
- Edit the Kubernetes service annotations to gateway-neg-two. 
- Let it sink in. At this point, 3 additional Network Endpoints are created with gateway-neg-two. 
- At this point, there are 6 network endpoints attached to the backend, 3 associated with gateway-neg and 3 associated with gateway-neg-two.
---
Additional Observation: If you delete the annotation from service, it deletes the Network endpoint group. My suspicion is that AutoNeg might not be sending request to deattach the network endpoints from the backend when there is a rename of the network endpoint group name.

I will try out changing the port name and check.. but I am expecting to see some weirdness there too. (Rename port name, instead of adding new ones)
